modules/040-node-manager/hooks/yandex_preemptibly_delete_preemtible_instances.go

```diff
@@ -148,6 +168,19 @@ func deleteMachines(input *go_hook.HookInput) error {
 		preemptibleMachineClassesSet.Add(ic.Name)
 	}
 
+	for _, nodeRaw := range input.Snapshots["nodes"] {
+		if nodeRaw == nil {
+			continue
+		}
+
+		node, ok := nodeRaw.(*Node)
+		if !ok {
+			return fmt.Errorf("failed to assert to *Node")
+		}
+
+		nodeCreationTimestampSet[node.Name] = node.CreationTimestamp
+	}
+
 	if preemptibleMachineClassesSet.Size() == 0 {
 		return nil
 	}
```

Comment:
I assume we can change an order of this functions. At first - check preemptibleMachineClassesSet size and return if 0. And only if `preemptibleMachineClassesSet` exists - iterate over nodes

184-186 lines move to 170

Reply:
Nice. We can do the same for `nodeCreationTimestampSet`, but that seems superfluous, since without Node objects, nothing will work in a cluster.